data/glsk/glsk-document-api/src/main/java/com/farao_community/farao/data/glsk/api/util/converters/GlskPointScalableConverter.java

```diff
@@ -123,25 +123,43 @@ private static Scalable convertMeritOrder(Network network, AbstractGlskPoint gls
                 .sorted(Comparator.comparingInt(AbstractGlskShiftKey::getMeritOrderPosition))
                 .map(glskShiftKey -> {
                     AbstractGlskRegisteredResource generatorRegisteredResource = Objects.requireNonNull(glskShiftKey.getRegisteredResourceArrayList()).get(0);
-                    String generatorId = generatorRegisteredResource.getGeneratorId();
-                    double incomingMaxP = generatorRegisteredResource.getMaximumCapacity().orElse(Double.MAX_VALUE);
-                    double incomingMinP = generatorRegisteredResource.getMinimumCapacity().orElse(-Double.MAX_VALUE);
-                    return (Scalable) Scalable.onGenerator(generatorId, incomingMinP, incomingMaxP);
+                    return getGeneratorScalableWithLimits(network, generatorRegisteredResource);
                 }).toArray(Scalable[]::new));
 
         Scalable downScalable = Scalable.stack(glskPoint.getGlskShiftKeys().stream()
                 .filter(abstractGlskShiftKey -> abstractGlskShiftKey.getMeritOrderPosition() < 0)
                 .sorted(Comparator.comparingInt(AbstractGlskShiftKey::getMeritOrderPosition).reversed())
                 .map(glskShiftKey -> {
                     AbstractGlskRegisteredResource generatorRegisteredResource = Objects.requireNonNull(glskShiftKey.getRegisteredResourceArrayList()).get(0);
-                    String generatorId = generatorRegisteredResource.getGeneratorId();
-                    double incomingMaxP = generatorRegisteredResource.getMaximumCapacity().orElse(Double.MAX_VALUE);
-                    double incomingMinP = generatorRegisteredResource.getMinimumCapacity().orElse(-Double.MAX_VALUE);
-                    return (Scalable) Scalable.onGenerator(generatorId, incomingMinP, incomingMaxP);
+                    return getGeneratorScalableWithLimits(network, generatorRegisteredResource);
                 }).toArray(Scalable[]::new));
         return Scalable.upDown(upScalable, downScalable);
     }
 
+    private static Scalable getGeneratorScalableWithLimits(Network network, AbstractGlskRegisteredResource generatorRegisteredResource) {
+        String generatorId = generatorRegisteredResource.getGeneratorId();
+        double incomingMaxP = generatorRegisteredResource.getMaximumCapacity().orElse(Double.MAX_VALUE);
+        double incomingMinP = generatorRegisteredResource.getMinimumCapacity().orElse(-Double.MAX_VALUE);
+        // Fixes some inconsistencies between GLSK and network that may raise an exception in
+        // PowSyBl when actually scaling the network.
+        // TODO: Solve this issue in PowSyBl framework.
```

Comment:
Should we create an issue in Powsybl repo ?

Reply:
I will discuss with them directly. Fix should not be too complicated, but we will have to wait for one more release, because release candidate has already been published, and its perimeter is now blocked.